File: api-frontend/apps/web-antd/src/api/system/menu.ts
import { requestClient } from '#/api/request';

export interface MenuRecord {
  id: number;
  tenant_id: number;
  parent_id: number;
  name: string;
  path: string;
  component: string;
  redirect: string;
  icon: string;
  title: string;
  hide_in_menu: number;
  hide_in_tab: number;
  hide_in_breadcrumb: number;
  keep_alive: number;
  frame_src: string;
  sort: number;
  status: number;
  permission_code: string;
  created_at: string;
  // 新增字段
  is_tenant: number;   // 1=租户可见 0=仅超管
  is_public: number;   // 1=公共
  type: number;        // 1=目录 2=菜单 3=页面 4=按钮
  children?: MenuRecord[];
}

export function getMenuList() {
  return requestClient.get<MenuRecord[]>('/menus');
}

export function getMenuTree() {
  return requestClient.get<MenuRecord[]>('/menus/tree');
}

export function createMenu(data: Partial<MenuRecord>) {
  return requestClient.post('/menus', data);
}

export function updateMenu(id: number, data: Partial<MenuRecord>) {
  return requestClient.put(`/menus/${id}`, data);
}

export function deleteMenu(id: number) {
  return requestClient.delete(`/menus/${id}`);
}

export function getUserMenus(userId: number) {
  return requestClient.get<MenuRecord[]>(`/menus/${userId}/menus`);
}

export function assignUserMenus(userId: number, menu_ids: number[]) {
  return requestClient.put(`/menus/${userId}/menus`, { menu_ids });
}

export interface ButtonRecord {
  title: string;
  name: string;
  permission_code: string;
  sort: number;
}

export function getMenuButtons(menuId: number) {
  return requestClient.get<MenuRecord[]>(`/menus/${menuId}/buttons`);
}

export function saveMenuButtons(menuId: number, buttons: ButtonRecord[]) {
  return requestClient.put(`/menus/${menuId}/buttons`, { buttons });
}

export function getMenuTreeWithButtons() {
  return requestClient.get<MenuRecord[]>('/menus/tree-with-buttons');
}

File: api-backend/internal/model/menu.go
package model

type Menu struct {
	BaseModel
	TenantID         uint   `gorm:"index;default:0" json:"tenant_id"`
	ParentID         uint   `gorm:"default:0" json:"parent_id"`
	Name             string `gorm:"size:50;not null" json:"name"`
	Path             string `gorm:"size:255" json:"path"`
	Component        string `gorm:"size:255" json:"component"`
	Redirect         string `gorm:"size:255" json:"redirect"`
	Icon             string `gorm:"size:50" json:"icon"`
	Title            string `gorm:"size:50" json:"title"`
	HideInMenu       int8   `gorm:"default:0" json:"hide_in_menu"`
	HideInTab        int8   `gorm:"default:0" json:"hide_in_tab"`
	HideInBreadcrumb int8   `gorm:"default:0" json:"hide_in_breadcrumb"`
	KeepAlive        int8   `gorm:"default:1" json:"keep_alive"`
	FrameSrc         string `gorm:"size:255" json:"frame_src"`
	Sort             int    `gorm:"default:0" json:"sort"`
	Status           int8   `gorm:"default:1" json:"status"`
	PermissionCode   string `gorm:"size:100" json:"permission_code"`
	Children         []Menu `gorm:"-" json:"children,omitempty"`
}

func (Menu) TableName() string {
	return "menus"
}

type MenuMeta struct {
	Title            string `json:"title"`
	Icon             string `json:"icon,omitempty"`
	HideInMenu       bool   `json:"hideInMenu,omitempty"`
	HideInTab        bool   `json:"hideInTab,omitempty"`
	HideInBreadcrumb bool   `json:"hideInBreadcrumb,omitempty"`
	KeepAlive        bool   `json:"keepAlive,omitempty"`
	FrameSrc         string `json:"frameSrc,omitempty"`
}

type MenuTree struct {
	ID        uint       `json:"id"`
	ParentID  uint       `json:"parentId"`
	Name      string     `json:"name"`
	Path      string     `json:"path"`
	Component string     `json:"component,omitempty"`
	Redirect  string     `json:"redirect,omitempty"`
	Meta      MenuMeta   `json:"meta"`
	Children  []MenuTree `json:"children,omitempty"`
}
